### ISSUE TITLE
Warning on java-library and spring-boot appears for all project tasks

### ISSUE THREAD

**Issue by pkubowicz** [autonomousapps/dependency-analysis-gradle-plugin#1270]
**Build scan link**
https://scans.gradle.com/s/7iev7o7xnp5m4/console-log

**Plugin version**
2.1.0

**Gradle version**
8.10.1

**JDK version**
17.0.12

**(Optional) Kotlin and Kotlin Gradle Plugin (KGP) version**
N/A

**(Optional) Android Gradle Plugin (AGP) version**
N/A

**(Optional) `reason` output for bugs relating to incorrect advice**
N/A

**Describe the bug**
1. I see warnings from dependency-analysis plugin although I don't execute it.
2. Plugins should refrain from doing unneeded things in configuration phase. This is inefficient from build performance point of view, as well as distracting (stealing attention when not used) from user experience point of view.
3. The warning appears twice, although I have just 1 project.

**To Reproduce**
1. Run `gradle classes` in

```gradle
plugins {
    id("com.autonomousapps.dependency-analysis") version "2.1.0"
    id("org.springframework.boot") version "3.3.1"
    `java-library`
}

repositories {
    mavenCentral()
}
```

Console output:
```
Calculating task graph as no cached configuration is available for tasks: classes

> Configure project :
(dependency analysis) You have both java-library and org.springframework.boot applied. You probably want java, not java-library.
(dependency analysis) You have both java-library and org.springframework.boot applied. You probably want java, not java-library.

BUILD SUCCESSFUL in 7s
```

**Expected behavior**
The warning may be fine, but only during execution phase, this is: only when buildHealth task is in the execution graph.

When the task is not in the execution graph, I shouldn't see any messages from this plugin.

**Additional context**
This bug is old, was present in 1.33.0 and some older versions.

One more question: what is the reason for warning about those plugins used together? What am I losing by continuing doing what I am warned about?

**Comment by autonomousapps** [comment]
Here's the code in question:
```kotlin
// Regardless of the fact that this is a "java-library" project, the presence of Spring
// Boot indicates an app project.
val dependencyAnalyzer = if (pluginManager.hasPlugin(SPRING_BOOT_PLUGIN)) {
  logger.warn(
    "(dependency analysis) You have both java-library and org.springframework.boot applied. You probably " +
      "want java, not java-library."
  )
```
So this is really just about reducing confusion. It also does this _per source set_, which is why you're seeing the warning twice. (Presumably once for main and once per test source set.)

This could be optimized.

**Comment by pkubowicz** [comment]
Back to the question I asked at the end. Is the warning valid? Is anything bad happening _in practice_ if you apply both plugins?

I'm asking because maybe the simplest fix is to remove the code above altogether, instead of trying to improve it.

**Comment by autonomousapps** [comment]
[The java-library plugin will add _api_ configurations for all source sets](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation), even though these are unnecessary and not used for application modules (which is what I think a "spring boot" module is, right?). So it makes your build less efficient, although in a fairly small way.

**Comment by autonomousapps** [comment]
To bring it back to the warning, I think you can switch your plugin from `java-library` to `java` and your build should still Just Work. If that's not the case, please let me know.

**Comment by pkubowicz** [comment]
I'm applying java-library with a convention plugin. Actually, I have the convention plugin 'kotlin' that includes the convention plugin 'jvm' that applies 'java-library'.

So to satisfy the irritating warning from the plugin, I would need to copy-paste non-trivial logic of 'jvm' convention to 'jvm-non-library' convention, then copy-paste 'kotlin' to 'kotlin-non-library'. Or add another level of 'inheritance'.

I don't see any justification of such a damage to my source code.

The consequence of my unnecessary use of java-library is that 2 subprojects out of 80 have an unused configuration declared. I don't expect this having any footprint other than statistical noise.

The warning from dependency-analysis 2.x is so noisy that I decided not to update from 1.x. It makes no sense that several teams see unnecessary warnings multiple times per day, from a plugin that is not used at all for daily work.

**Comment by autonomousapps** [comment]
I don't see any reason for you to be rude about this. I thought we were having a productive discussion. Also, you're weirdly wrong: that warning has existed for 3 years and has nothing to do with the plugin's v2 release.

You are of course under no obligation to use this plugin. It's free and I work on it in a volunteer capacity.

This is no longer worth my time. Closing.

**Comment by pkubowicz** [comment]
Before 2.x, I was able to get rid of the warning by disabling auto-applying of the plugin. The property for auto-applying was removed in 2.x. Am I forbidden from raising a bug just because it's 3 years old?

Not sure what you consider 'rude' in my messages. As for the productive discussion, I did not feel we were having it, as you seemed to focus on giving me unsolicited advice on how my code should look like, while avoiding discussion over objective facts pointed (that Gradle docs say plugins should not do work in the configuration phase, that neither Gradle nor Spring documentation discourage the combination what the plugins try to discourage etc.). I never gave requirements how your code should look like (note 'may be fine' and 'might the fix is'), which was met with quite a different approach on my code from your side. Take my apologies if this was not your intention at all.
